code/modules/antagonists/cult/cult_comms.dm

```diff
@@ -79,6 +79,101 @@
 			var/link = FOLLOW_LINK(M, user)
 			to_chat(M, "[link] [my_message]")
 
+///a button for requesting cultists come to your location, lasts for 3 minutes (during which time no one else can request assistance)
+/datum/action/innate/cult/request_aid
+	name = "Request Assistance"
+	desc = "Request a number of cultists of your choosing to your location."
+	button_icon_state = "assistance"
+	check_flags = AB_CHECK_HANDS_BLOCKED|AB_CHECK_IMMOBILE|AB_CHECK_CONSCIOUS
+	var/cooldown = 0
+	var/base_cooldown = 180 SECONDS
+
+/datum/action/innate/cult/request_aid/Activate()
+	var/datum/antagonist/cult/team = owner.mind.has_antag_datum(/datum/antagonist/cult,TRUE)
+	if(world.time < cooldown)
+		to_chat(owner, span_warning("You need to wait longer before requesting assistance again!"))
+		return
+	if(team.cult_team.need_of_help) //check to make sure no one else has already requested assistance
+		to_chat(owner, span_warning("Someone else is currently requesting assistance!"))
+		return
+	var/number = tgui_alert(owner, "How many cultists do you need?",, list("0", "1", "2", "3"))
+	if(number == "0")
+		to_chat(owner, "You decide not to request help.")
+		return
+	team.cult_team.need_of_help = owner
```

Comment:
This needs to be a weakref, especially if it can last for 3 minutes (which means a GC run can happen in between, IIRC).

You can ask me for help and/or look at other code with `WEAKREF` to learn how to use them.